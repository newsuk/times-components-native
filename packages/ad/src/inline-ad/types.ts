import { ParagraphContent } from "./domain-types";

export type Line = {
  text: string;
};
export type TextLayoutEvent = { lines: Line[] };
export type ColumnContents = ParagraphContent[];
export interface PageColumn {
  byline?: {
    height: number;
  };
  image?: {
    height: number;
    margin: number;
  };
  contents: ColumnContents;
  contentHeight: number;
}

export interface ContentMeasurements {
  contents: {
    lines: { [key: string]: Line[] };
    heights: { [key: string]: number };
  };
}

export interface ContentParameters {
  contentWidth: number;
  contentHeight: number;
  contentLineHeight: number;
}

// export interface FontScale {
//   fontSize: number;
//   lineHeight: number;
// }
