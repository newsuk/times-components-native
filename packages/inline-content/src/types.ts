import { ParagraphContent } from "@times-components-native/types";

export type Line = {
  text: string;
};
export type TextLayoutEvent = { lines: Line[] };
export type ChunkContents = ParagraphContent[];

export interface ContentParameters {
  contentWidth: number;
  contentHeight: number;
  contentLineHeight: number;
  itemWidth: number;
}

export type SkeletonProps = Record<string, unknown>;

interface BaseInlineContentProps {
  adConfig: Record<string, unknown>;
  defaultFont: { lineHeight: number };
  display: string;
  height: number;
  inlineContent: ParagraphContent[];
  narrowContent: boolean;
  originalName: string;
  skeletonProps: SkeletonProps;
  slotName: string;
  width: number;
}

export interface InlineArticleImageProps extends BaseInlineContentProps {
  caption: string;
  credits: string;
  imageIndex: number;
  narrowContent: boolean;
  onImagePress: () => void;
  ratio: string;
  relativeWidth: number;
  relativeHeight: number;
  relativeHorizontalOffset: number;
  relativeVerticalOffset: number;
  url: string;
}

export interface InlinePullQuoteProps extends BaseInlineContentProps {
  caption: { name: string; text: string; twitter: string };
  children: [{ string: string }];
  onTwitterLinkPress: () => void;
}

export type InlineContentProps =
  | BaseInlineContentProps
  | InlineArticleImageProps
  | InlinePullQuoteProps;

export type ArticleImageProps = {
  captionOptions: {
    caption: string;
    credits: string;
  };
  onImagePress: () => void;
  images: [];
  imageOptions: {
    display: string;
    ratio: string;
    index: number;
    uri: string;
    relativeWidth: number;
    relativeHeight: number;
    relativeHorizontalOffset: number;
    relativeVerticalOffset: number;
    narrowContent: boolean;
  };
  originalName: string;
};

export type PullQuoteProps = {
  caption: string;
  children: [{ string: string }];
  onTwitterLinkPress: () => void;
  originalName: string;
  text: string;
  twitter: string;
  width: number;
};

export type InlineItemProps = ArticleImageProps | PullQuoteProps;
