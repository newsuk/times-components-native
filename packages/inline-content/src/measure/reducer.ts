import { Reducer } from "react";

import { Measurements } from "@times-components-native/types";
import { Line } from "../types";

export const initialState: Measurements = {
  contents: {
    lines: {},
    heights: {},
  },
  itemHeight: null,
};

interface SetInlineContentHeightAction {
  type: "SET_INLINE_CONTENT_HEIGHT";
  id: string;
  payload: number;
}

interface SetInlineContentLinesAction {
  type: "SET_INLINE_CONTENT_LINES";
  id: string;
  payload: Line[];
}

interface SetInlineItemHeightAction {
  type: "SET_INLINE_ITEM_HEIGHT";
  height: number;
}

export type Action =
  | SetInlineContentHeightAction
  | SetInlineContentLinesAction
  | SetInlineItemHeightAction;

export const reducer: Reducer<Measurements, Action> = (state, action) => {
  switch (action.type) {
    case "SET_INLINE_CONTENT_HEIGHT":
      return {
        ...state,
        contents: {
          ...state.contents,
          heights: { ...state.contents.heights, [action.id]: action.payload },
        },
      };
    case "SET_INLINE_CONTENT_LINES":
      return {
        ...state,
        contents: {
          ...state.contents,
          lines: { ...state.contents.lines, [action.id]: action.payload },
        },
      };
    case "SET_INLINE_ITEM_HEIGHT":
      return {
        ...state,
        itemHeight: action.height,
      };
    default:
      return state;
  }
};
