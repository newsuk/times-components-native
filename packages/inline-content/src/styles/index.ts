import { StyleSheet, ViewStyle } from "react-native";

import { spacing, tabletWidth } from "@times-components-native/styleguide";

interface Styles {
  container: ViewStyle;
  inlineItemContainer: ViewStyle;
}

const styles: Styles = {
  container: {
    flex: 1,
    flexDirection: "row",
    width: tabletWidth,
  },
  inlineItemContainer: {
    marginLeft: spacing(2),
    marginBottom: spacing(2),
  },
};

export default StyleSheet.create<Styles>(styles);
