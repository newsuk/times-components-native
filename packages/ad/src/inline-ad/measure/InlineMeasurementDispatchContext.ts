/* istanbul ignore file */

import { createContext, useContext } from "react";

import { Action } from "./reducer";

export const InlineMeasurementDispatch = createContext<React.Dispatch<Action>>(
  () => ({}),
);
export const useInlineMeasurementDispatchContext = () =>
  useContext(InlineMeasurementDispatch);
