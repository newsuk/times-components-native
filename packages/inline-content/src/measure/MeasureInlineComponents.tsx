import React, { memo } from "react";
import { View } from "react-native";

import ArticleImage from "@times-components-native/article-image";
import renderTrees from "@times-components-native/markup-forest";
import getRenderers from "@times-components-native/article-skeleton/src/article-body/article-body-row.js";
import { ParagraphContent } from "@times-components-native/types";
import { useInlineMeasurementDispatchContext } from "./InlineMeasurementDispatchContext";
import { ArticleImageProps, SkeletonProps } from "../types";

interface Props {
  content: ParagraphContent;
  skeletonProps: SkeletonProps;
}

export const MeasureContent: React.FC<Props> = memo(
  ({ content, skeletonProps }) => {
    const dispatch = useInlineMeasurementDispatchContext();

    const inlineContent = {
      ...content,
      attributes: { ...content.attributes, inline: true },
    };

    const { data, isTablet, narrowContent, scale } = skeletonProps;

    return (
      <View
        onLayout={(event) => {
          const height = Math.floor(event.nativeEvent.layout.height);
          dispatch({
            type: "SET_INLINE_CONTENT_HEIGHT",
            id: inlineContent.id!,
            payload: height,
          });
        }}
      >
        {renderTrees(
          [inlineContent],
          getRenderers({
            data,
            interactiveConfig: {},
            onLinkPress: null,
            onTwitterLinkPress: null,
            onVideoPress: null,
            onImagePress: null,
            isTablet,
            adConfig: {},
            images: [],
            dropcapsDisabled: true,
            dropCapFont: "dropCap",
            scale,
            analyticsStream: null,
            narrowContent,
            onParagraphTextLayout: (event: any) => {
              const lines = event.nativeEvent.lines;
              dispatch({
                type: "SET_INLINE_CONTENT_LINES",
                id: inlineContent.id!,
                payload: lines,
              });
            },
          }),
        )}
      </View>
    );
  },
);

export const MeasureItem: React.FC<{
  itemProps: ArticleImageProps;
  width: number;
}> = memo(({ itemProps, width }) => {
  const dispatch = useInlineMeasurementDispatchContext();

  if (!itemProps || !width)
    dispatch({
      type: "SET_INLINE_ITEM_HEIGHT",
      height: 0,
    });

  return (
    <View
      style={{ width }}
      onLayout={(event) => {
        const height = event.nativeEvent.layout.height;
        dispatch({
          type: "SET_INLINE_ITEM_HEIGHT",
          height,
        });
      }}
    >
      <ArticleImage {...itemProps} />
    </View>
  );
});
