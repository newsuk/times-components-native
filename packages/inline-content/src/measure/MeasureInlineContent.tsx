import React, { useReducer } from "react";
import { ScrollView, StyleSheet, View } from "react-native";

import { Measurements, ParagraphContent } from "@times-components-native/types";
import { InlineMeasurementDispatch } from "./InlineMeasurementDispatchContext";
import { MeasureItem, MeasureContent } from "./MeasureInlineComponents";
import { initialState, reducer } from "./reducer";
import {
  ContentParameters,
  InlineItemProps,
  Line,
  SkeletonProps,
} from "../types";

interface Props {
  content: ParagraphContent[];
  contentParameters: ContentParameters;
  itemProps: InlineItemProps;
  renderMeasuredContents: (contentMeasurements: Measurements) => any;
  skeletonProps: SkeletonProps;
}

const allContentMeasured = (
  contentToMeasure: ParagraphContent[],
  heights: { [key: string]: number },
  lines: { [key: string]: Line[] },
) => {
  return (
    contentToMeasure.filter((c) => heights[c.id!] && lines[c.id!]).length ===
    contentToMeasure.length
  );
};

export const InnerMeasureInlineContent: React.FC<
  Props & { measurementState: Measurements }
> = ({
  content,
  contentParameters,
  itemProps,
  measurementState,
  renderMeasuredContents,
  skeletonProps,
}) => {
  const {
    contents: { lines, heights },
  } = measurementState;
  if (
    allContentMeasured(content, heights, lines) &&
    (!itemProps || measurementState.itemHeight !== null)
  ) {
    return renderMeasuredContents(measurementState);
  }

  return (
    <ScrollView style={styles.renderOffscreen}>
      {itemProps && (
        <MeasureItem
          itemProps={itemProps}
          width={contentParameters.itemWidth}
        />
      )}
      <View
        style={{
          width: contentParameters.contentWidth,
        }}
      >
        {content.map((contentItem) => (
          <MeasureContent
            key={`InlineContentMeasuringView:${contentItem.id}`}
            skeletonProps={skeletonProps}
            content={contentItem}
          />
        ))}
      </View>
    </ScrollView>
  );
};

/* istanbul ignore next */
export const MeasureInlineContent: React.FC<Props> = (props) => {
  const [measurementState, measurementDispatch] = useReducer(
    reducer,
    initialState,
  );

  return (
    <InlineMeasurementDispatch.Provider value={measurementDispatch}>
      <InnerMeasureInlineContent
        {...props}
        measurementState={measurementState}
      />
    </InlineMeasurementDispatch.Provider>
  );
};

const styles = StyleSheet.create({
  renderOffscreen: {
    transform: [{ translateX: -1000 }],
  },
});
