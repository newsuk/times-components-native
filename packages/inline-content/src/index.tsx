import React, { useCallback, FC } from "react";
import { View, Text } from "react-native";

import { ParagraphContent } from "@times-components-native/types";
import {
  colours,
  spacing,
  tabletWidth,
  getNarrowArticleBreakpoint,
} from "@times-components-native/styleguide";
import { render } from "@times-components-native/markup-forest";
import {
  renderers,
  Gutter,
  ErrorBoundary,
} from "@times-components-native/article-skeleton";
import { useResponsiveContext } from "@times-components-native/responsive";
import Context from "@times-components-native/context";
import ArticleImage from "@times-components-native/article-image";
import PullQuote from "@times-components-native/pull-quote";

import { chunkInlineContent } from "./utils/chunkInlineContent";
import { MeasureInlineContent } from "./measure/MeasureInlineContent";

import styles from "./styles";
import { ArticleImageProps, SkeletonProps } from "./types";

const assignWithId = (height: number) => (
  content: ParagraphContent,
  idx: number,
): ParagraphContent => {
  return {
    ...content,
    id: `${idx}-${height}`, //suffixing the height ensures that we re-measure the content if the orientation changes - and that we don't unnecessarily re-measure if orientation changes back
  };
};

export const renderItemComponent = (itemProps) => {
  const { children, originalName, width } = itemProps;
  if (originalName === "image") {
    return <ArticleImage {...itemProps} />;
  }
  if (originalName === "pullQuote") {
    const itemContent = children?.[0]?.string;
    return (
      <Context.Consumer>
        {({
          theme: { pullQuoteFont, sectionColour = colours.section.default },
        }) => (
          <View style={{ width }}>
            <PullQuote
              {...itemProps}
              font={pullQuoteFont}
              quoteColour={sectionColour}
            >
              {itemContent}
            </PullQuote>
          </View>
        )}
      </Context.Consumer>
    );
  }
};

interface Props {
  adConfig: Record<string, unknown>;
  defaultFont: { lineHeight: number };
  display: string;
  height: number;
  inlineContent: ParagraphContent[];
  originalName: string;
  skeletonProps: SkeletonProps;
  slotName: string;
  width: number;
}

const InlineContent: FC<Props> = (props) => {
  const {
    defaultFont,
    inlineContent,
    narrowContent,
    originalName,
    skeletonProps,
  } = props;

  const { windowWidth } = useResponsiveContext();

  const renderChild = render(
    // @ts-ignore
    renderers({ dropcapsDisabled: true, ...skeletonProps }),
  );

  const Child = useCallback(({ item, index }, inline = false) => {
    item.attributes = { ...item.attributes, inline };

    return (
      <Gutter style={{ overflow: "hidden" }}>
        <ErrorBoundary>
          {renderChild(item, index.toString(), index)}
        </ErrorBoundary>
      </Gutter>
    );
  }, []);

  const renderItem = (inline: boolean) => (
    item: ParagraphContent,
    index: number,
  ) => Child({ item, index }, inline);

  const { lineHeight } = defaultFont;

  const availableWidth = Math.min(
    windowWidth,
    narrowContent
      ? getNarrowArticleBreakpoint(windowWidth).content
      : tabletWidth,
  );

  const inlineItemWidth = availableWidth * 0.35;
  const inlineContentWidth = availableWidth - inlineItemWidth;

  let itemProps: ArticleImageProps;

  console.log("OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO", originalName);

  if (originalName === "image") {
    const {
      caption,
      credits,
      display,
      imageIndex,
      // narrowContent,
      onImagePress,
      ratio,
      relativeWidth,
      relativeHeight,
      relativeHorizontalOffset,
      relativeVerticalOffset,
      url,
    } = props;

    itemProps = {
      captionOptions: {
        caption,
        credits,
      },
      onImagePress,
      originalName,
      images: [],
      imageOptions: {
        display,
        ratio,
        index: imageIndex,
        uri: url,
        relativeWidth,
        relativeHeight,
        relativeHorizontalOffset,
        relativeVerticalOffset,
        narrowContent,
      },
    };
  }

  // if (originalName === "pullQuote") {
  const {
    caption: { name, text, twitter },
    children,
    onTwitterLinkPress,
  } = props;

  itemProps = {
    caption: name,
    children,
    onTwitterLinkPress,
    originalName,
    text,
    twitter,
    width: inlineItemWidth,
  };
  // }

  // const [ratioWidth, ratioHeight] = ratio.split(":");
  // const aspectRatio = ratioWidth / ratioHeight;
  // const inlineItemHeight = inlineItemWidth / aspectRatio;
  // const inlineContentHeight = inlineItemHeight + spacing(2); // <<<<<<<<<<<<<<<< TODO!!!!!!!!!!!!!!

  // const adHeaderHeight = spacing(4);
  // const adHorizontalSpacing = 21;
  // const adMarginBottom = spacing(2);
  // const adContainerHeight = height + adHeaderHeight;
  // const adContainerHeightPlusMargin = adContainerHeight + adMarginBottom;
  // const adContainerWidth = width + adHorizontalSpacing;
  // const contentWidth = tabletWidth - adContainerWidth;

  // const contentHeight = adContainerHeightPlusMargin;

  const paragraphs = inlineContent
    .filter((c) => c.name === "paragraph")
    .map(assignWithId(inlineContentWidth));

  const contentParameters = {
    contentWidth: inlineContentWidth,
    // contentHeight: inlineContentHeight,
    contentHeight: 0,
    contentLineHeight: lineHeight,
    itemWidth: inlineItemWidth,
  };

  return (
    <MeasureInlineContent
      content={paragraphs}
      contentParameters={contentParameters}
      itemProps={itemProps}
      skeletonProps={skeletonProps}
      renderMeasuredContents={(contentMeasurements) => {
        const { chunks, currentInlineContentHeight } = chunkInlineContent(
          paragraphs,
          contentMeasurements,
          contentParameters,
        );

        // const itemHeight =
        //   contentMeasurements.itemHeight || inlineContentHeight || 0;
        const itemHeight = contentMeasurements.itemHeight || 0;

        const requiredInlineContentHeight = Math.max(
          currentInlineContentHeight,
          itemHeight,
        );

        const chunkedInlineContent = chunks[0] || [];
        const chunkedOverflowContent = chunks[1] || [];

        return (
          <>
            <View
              style={[
                styles.container,
                { height: requiredInlineContentHeight },
              ]}
            >
              <View
                style={[
                  narrowContent
                    ? styles.inlineItemNarrowContainer
                    : styles.inlineItemContainer,
                  { width: inlineItemWidth, height: itemHeight },
                ]}
              >
                {renderItemComponent(itemProps)}
                {/* <ItemComponent {...itemProps} /> */}
                {/* {Component} */}
              </View>
              <View
                style={{
                  height: requiredInlineContentHeight,
                  width: inlineContentWidth,
                }}
              >
                {chunkedInlineContent.map(renderItem(true))}
              </View>
            </View>
            {chunkedOverflowContent.map(renderItem(false))}
          </>
        );
      }}
    />
  );
};

export default InlineContent;
