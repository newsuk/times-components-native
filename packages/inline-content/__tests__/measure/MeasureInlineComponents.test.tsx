import React from "react";
import { Text, View } from "react-native";
import ReactTestRenderer from "react-test-renderer";

import { ParagraphContent } from "@times-components-native/types";
import { InlineMeasurementDispatch } from "../../src/measure/InlineMeasurementDispatchContext";
import {
  MeasureContent,
  MeasureItem,
} from "../../src/measure/MeasureInlineComponents";
import { InlineItemProps } from "../../src/types";

jest.mock("@times-components-native/ad", () => "Ad");
jest.mock("@times-components-native/article-image", () => "ArticleImage");
jest.mock("@times-components-native/pull-quote", () => "PullQuote");
jest.mock("react-native-image-zoom-viewer", () => "ImageZoomView");

export const createParagraphWithText = (text: string): ParagraphContent => ({
  id: "some-paragraph-id",
  name: "paragraph",
  children: [{ name: "text", children: [], attributes: { value: text } }],
});

const skeletonProps = {
  data: [],
  isTablet: true,
  narrowContent: false,
  scale: 1,
};

describe("MeasureContent", () => {
  const paragraph = createParagraphWithText("abc");
  it("renders the content", () => {
    const renderer = ReactTestRenderer.create(
      <MeasureContent content={paragraph} skeletonProps={skeletonProps} />,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("records the measurement", () => {
    const mock = jest.fn();
    const renderer = ReactTestRenderer.create(
      <InlineMeasurementDispatch.Provider value={mock}>
        <MeasureContent content={paragraph} skeletonProps={skeletonProps} />
      </InlineMeasurementDispatch.Provider>,
    );

    renderer.root.findByType(View).props["onLayout"]({
      nativeEvent: { layout: { height: 10 } },
    });

    expect(mock).toHaveBeenCalledWith({
      payload: 10,
      id: paragraph.id,
      type: "SET_INLINE_CONTENT_HEIGHT",
    });
  });

  it("records the captured lines", () => {
    const mock = jest.fn();
    const renderer = ReactTestRenderer.create(
      <InlineMeasurementDispatch.Provider value={mock}>
        <MeasureContent content={paragraph} skeletonProps={skeletonProps} />
      </InlineMeasurementDispatch.Provider>,
    );

    renderer.root.findAllByType(Text)[0].props["onTextLayout"]({
      nativeEvent: { lines: [{ text: "abc" }] },
    });

    expect(mock).toHaveBeenCalledWith({
      payload: [{ text: "abc" }],
      id: paragraph.id,
      type: "SET_INLINE_CONTENT_LINES",
    });
  });
});

describe("MeasureItem", () => {
  const pullQuoteItemProps = {
    caption: "some caption",
    children: [{ string: "some pull quote text" }],
    onTwitterLinkPress: () => {
      return;
    },
    originalName: "pullQuote",
    text: "some text",
    twitter: "some twitter text",
    width: 231,
  } as InlineItemProps;

  it("renders a pull quote", () => {
    const renderer = ReactTestRenderer.create(
      <MeasureItem itemProps={pullQuoteItemProps} width={231} />,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  const imageItemProps = {
    captionOptions: {
      caption: "some caption",
      credits: "some credits",
    },
    images: [],
    imageOptions: {
      display: "inline",
      index: 0,
      narrowContent: false,
      ratio: "16:9",
      relativeHeight: 900,
      relativeHorizontalOffset: 0,
      relativeVerticalOffset: 0,
      relativeWidth: 1600,
      uri: "some image uri",
    },
    onImagePress: () => {
      return;
    },
    originalName: "image",
  } as InlineItemProps;

  it("renders an inline image", () => {
    const renderer = ReactTestRenderer.create(
      <MeasureItem itemProps={imageItemProps} width={231} />,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("records the measurement of an item", () => {
    const mock = jest.fn();
    const renderer = ReactTestRenderer.create(
      <InlineMeasurementDispatch.Provider value={mock}>
        <MeasureItem itemProps={pullQuoteItemProps} width={231} />,
      </InlineMeasurementDispatch.Provider>,
    );

    renderer.root.findByType(View).props["onLayout"]({
      nativeEvent: { layout: { height: 300 } },
    });

    expect(mock).toHaveBeenCalledWith({
      height: 300,
      type: "SET_INLINE_ITEM_HEIGHT",
    });
  });
});
