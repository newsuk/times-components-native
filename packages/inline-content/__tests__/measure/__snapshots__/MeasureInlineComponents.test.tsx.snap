// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeasureContent renders the content 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
        },
        Object {
          "marginBottom": 20,
        },
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    <View>
      <Text
        allowFontScaling={false}
        onTextLayout={[Function]}
        selectable={true}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "TimesDigitalW04",
            "fontSize": 36,
            "lineHeight": 52,
          }
        }
        testID="paragraph"
      >
        <Text>
          abc
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`MeasureItem renders a pull quote 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "width": 231,
    }
  }
>
  <PullQuote
    caption="some caption"
    onTwitterLinkPress={[Function]}
    originalName="pullQuote"
    text="some text"
    twitter="some twitter text"
    width={231}
  >
    some pull quote text
  </PullQuote>
</View>
`;

exports[`MeasureItem renders an inline image 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "width": 231,
    }
  }
>
  <ArticleImage
    captionOptions={
      Object {
        "caption": "some caption",
        "credits": "some credits",
      }
    }
    imageOptions={
      Object {
        "display": "inline",
        "index": 0,
        "narrowContent": false,
        "ratio": "16:9",
        "relativeHeight": 900,
        "relativeHorizontalOffset": 0,
        "relativeVerticalOffset": 0,
        "relativeWidth": 1600,
        "uri": "some image uri",
      }
    }
    images={Array []}
    onImagePress={[Function]}
    originalName="image"
  />
</View>
`;
